pkg/controller/resourcegraphdefinition/controller_cleanup.go

```diff
@@ -46,10 +56,16 @@ func (r *ResourceGraphDefinitionReconciler) cleanupResourceGraphDefinition(ctx c
 	}
 	// cleanup CRD
 	crdName := extractCRDName(group, rgd.Spec.Schema.Kind)
-	if err := r.cleanupResourceGraphDefinitionCRD(ctx, crdName); err != nil {
+	completed, err := r.cleanupResourceGraphDefinitionCRD(ctx, crdName)
+	if err != nil {
 		return fmt.Errorf("failed to cleanup CRD %s: %w", crdName, err)
 	}
 
+	if !completed {
+		log.V(1).Info("CRD deletion in progress, requeuing", "crd", crdName)
+		return requeue.NeededAfter(nil, crdDeletionRequeueDuration)
+	}
```

Comment:
I sense this has some overlap with KREP-004 (#763 )
with KREP-004 we wouldn't need to manange requeues on our own. RGD reconciliation would be re-triggered as the CRD got deleted